File: Data Structures & Algorithms/eating-bananas/submission-1.cpp
class Solution {
public:
    int findmax( vector<int>& v ) {
    int maxi = INT_MIN;
    int n = v.size();
    for( int i=0; i<n; i++ ) {
        maxi = max(maxi,v[i]);
    }
    return maxi;
}

int calculatetotalhours( vector<int> & v , int hourly ) {
    int totalH = 0;
    int n = v.size();
    for( int i=0; i<n; i++ ) {
        totalH += ceil( (double)v[i] / (double) hourly );
    }
    return totalH;
}

int minEatingSpeed(vector<int>& v, int h) {
    int low = 1;
    int high = findmax(v);
    while( low <= high ) {
        int mid = low + ( high - low ) /2;
        int totalH = calculatetotalhours(v,mid);
        if( totalH <= h ) {
            high = mid -1;
        }
        else {
            low = mid + 1;
        }
    }
    return low;
}

};

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-1.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        
       int sum = 0 ,  maxi = 0;

       for( int i=0; i<nums.length-1; i++ ) {
            sum = 0;
            sum += nums[i];
            for( int j=i; j<nums.length-1; j++ ) {
                if( nums[j] < nums[j+1] ) {
                    sum += nums[j+1];
                }
                else break;
                maxi = Math.max( maxi , sum );
            }
       }

       if( maxi == 0 ) {
        return nums[0];
       }
       return maxi;

    }
}